subsys/fs/nvs/nvs.c

```diff
@@ -418,10 +457,18 @@ static int nvs_flash_wrt_entry(struct nvs_fs *fs, uint16_t id, const void *data,
 
 	nvs_ate_crc8_update(&entry);
 
-	rc = nvs_flash_data_wrt(fs, data, len);
+	rc = nvs_flash_data_wrt(fs, data, len, true);
```

Comment:
This should be true only when CONFIG_NVS_DATA_CRC is enabled right ?

Reply:
Currently the CRC computation is disabled when the `CONFIG_NVS_DATA_CRC` is disabled directly in the `nvs_flash_data_wrt()` function, to avoid putting `ifdef` on each `nvs_flash_data_wrt()` call (see https://github.com/zephyrproject-rtos/zephyr/pull/73436/files#diff-d253a798f77267d97d9d96ebf25162efa0338bba3ed8e3a261439477a57444b7R116). Is it OK for you ?

Reply:
@RICCIARDI-Adrien you can define something like this to avoid ifdefs
#define data_crc_enabled IS_ENABLED(CONFIG_NVS_DATA_CRC)
Then use it as argument

Reply:
The code has been reworked based on this suggestion to avoid several `ifdef`.